include/swift/AST/DiagnosticsCommon.def

```diff
@@ -95,6 +95,9 @@ NOTE(profile_read_error,none,
 ERROR(generic_signature_not_minimal,none,
       "generic requirement '%0' is redundant in %1", (StringRef, StringRef))
 
+WARNING(protocol_extension_redundant_requirement,none,
+"requirement of '%1': '%2' is redundant in an extension of '%0'", (StringRef, StringRef, StringRef))
```

Comment:
Nitpick: Can you indent this and reflow the parameters?
Also, I think I'd prefer `to` over `:` in the diagnostic text.

Reply:
> Can you indent this and reflow the parameters?
Sure, not sure why the indentation keeps messing up. Something's definitely wrong with my editor. 🤔

> I think I'd prefer to over : in the diagnostic text.
Hmm, I chose `:` because the requirement is `Self: <Protocol Name>` so I wanted to mirror the code, but it's the same thing really, so I guess `to` is fine as well!

I have made the changes now!

Reply:
If you're curious, we have a great document about how we like to structure diagnostic text at [docs/Diagnostics.md](https://github.com/apple/swift/blob/master/docs/Diagnostics.md)

Reply:
Thanks a lot, I'll give it a read 👍